src/CSLOLUtils.cpp

```diff
@@ -118,10 +118,84 @@ bool CSLOLUtils::isUnnecessaryAdmin() {
     }
     return result;
 }
-#else
-// TODO: macos implementation
-QString CSLOLUtils::detectGamePath() { return ""; }
+#elif __APPLE__
+#    include <libproc.h>
+#    include <string.h>
+
+#    include <filesystem>
+#    include <iostream>
+
+QString CSLOLUtils::detectGamePath() {
+    pid_t *pid_list;
+
+    //
+    //  Get necessary size for `pid_list`
+    //
+    int bytes = proc_listallpids(nullptr, 0);
+
+    //
+    //  Something went wrong trying
+    //  to find all the processes.
+    //
+    if (bytes <= 0) {
+        return "";
+    }
+
+    pid_list = (pid_t *)malloc(bytes * sizeof(pid_t));
+
+    bytes = proc_listallpids(pid_list, bytes * sizeof(pid_t));
+
+    for (int i = 0; i < bytes / sizeof(pid_list[0]); i++) {
+        struct proc_bsdinfo proc;
+
+        //
+        //  Info about this PID.
+        //
+        int st = proc_pidinfo(pid_list[i], PROC_PIDTBSDINFO, 0, &proc, PROC_PIDTBSDINFO_SIZE);
+
+        //
+        //  We're looking for a LeagueClient PID.
+        //
+        if (strcmp(proc.pbi_name, "LeagueClient") == 0) {
+            char game_path[PROC_PIDPATHINFO_MAXSIZE];
+
+            //
+            //  Find the parent path
+            //  of the LeagueClient PID.
+            //
+            int len = proc_pidpath(proc.pbi_pid, game_path, sizeof(game_path));
+
+            //
+            //  Something went wrong.
+            //
+            if (len == 0)
+                qDebug("UNABLE TO LOCATE PARENT PATH OF LeagueClient PID!");
+
+            else {
+                //
+                //  `strtok()` will return the original
+                //  string we passed in, but it'll be modified.
+                //
+                //  We'll have the first part of the “split”,
+                //  in this case that's `/Applications/League of Legends`,
+                //  so we need to add the rest of the path to it.
+                //
+                //  This modification will persist for `game_path`.
+                //
+                strcat(strtok(game_path, "."), ".app/Contents/LoL/Game");
+
+                //
+                //  Ensure that /Applications/League of Legends.app/Contents/LoL/Game
+                //  is a valid directory.
+                //
+                if (std::filesystem::is_directory(game_path)) return game_path;
```

Comment:
```suggestion
                if (QFileInfo info(game_path); info.isDir()) {
                    free(pid_list);
                    return game_path;
                }
```
Avoid using std::filesystem in manager as it links to filesystem library on macos.
We are on Qt which implements all relevant filesystem APIs for us anyways, see example:
https://doc.qt.io/qt-6/qfileinfo.html#isDir

Also we need to free memory here.

Reply:
:+1: